pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2021 kubeflow.org.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("ServiceReconciler")
+
+//ServiceReconciler is the struct of Raw K8S Object
+type ServiceReconciler struct {
+	client          client.Client
+	scheme          *runtime.Scheme
+	Service         *corev1.Service
+	componentExt    *v1beta1.ComponentExtensionSpec
+	componentStatus v1beta1.ComponentStatusSpec
+}
+
+func NewServiceReconciler(client client.Client,
+	scheme *runtime.Scheme,
+	componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *ServiceReconciler {
+	return &ServiceReconciler{
+		client:          client,
+		scheme:          scheme,
+		Service:         createService(componentMeta, componentExt, podSpec, componentStatus),
+		componentExt:    componentExt,
+		componentStatus: componentStatus,
+	}
+}
+
+func createService(componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *corev1.Service {
+	var port int
+	if podSpec.Containers != nil && podSpec.Containers[0].Ports != nil {
+		port = int(podSpec.Containers[0].Ports[0].ContainerPort)
+	} else {
+		port, _ = strconv.Atoi(constants.InferenceServiceDefaultHttpPort)
+	}
+	service := &corev1.Service{
+		ObjectMeta: componentMeta,
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": constants.GetRawServiceLabel(componentMeta.Name),
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name: componentMeta.Name,
+					Port: int32(port),
```

Comment:
I think here we might use port 80 otherwise when user is calling from in cluster they need to know the internal port the container exposes, we can do the mapping from 80 -> user port here.